File: Models/TenTen.cs
namespace MeTenTenMaui.Models
{
    public class TenTen
    {
        public int Id { get; set; }
        public string? FirebaseKey { get; set; } // Firebase Realtime DB key
        public string Content { get; set; } = string.Empty;
        public DateTime CreatedAt { get; set; }
        public DateTime? UpdatedAt { get; set; }
        public int UserId { get; set; }
        public string UserName { get; set; } = string.Empty;
        public string TopicId { get; set; } = string.Empty; // Firebase Key 또는 int ID
        public string TopicSubject { get; set; } = string.Empty;
        public bool IsReadByPartner { get; set; }
        public DateTime? ReadByPartnerAt { get; set; }
        public bool IsEncrypted { get; set; } = true; // v1.2부터는 기본 true
        public string EncryptionType { get; set; } = "personal"; // "personal" or "shared"
    }
}

File: webapp/src/types/tenten.ts
/**
 * TenTen 관련 타입 정의
 */

import { FirebaseEntity, EncryptionType } from './common';

export interface TenTen extends FirebaseEntity {
  id: number; // 로컬 ID
  content: string; // 암호화된 내용 (Base64)
  topicId: string; // 연결된 Topic의 firebaseKey
  userId: string; // Firebase UID
  userName: string; // 작성자 이름
  topicSubject: string; // Topic 제목 (캐싱용)
  isEncrypted: boolean; // 암호화 여부
  encryptionType: EncryptionType; // "personal" or "shared"
  isReadByPartner: boolean; // 배우자가 읽었는지 여부
  readByPartnerAt?: string; // 배우자가 읽은 시간
}

export interface CreateTenTenRequest {
  content: string; // 평문
  topicId: string;
  encryptionType: EncryptionType;
}

export interface UpdateTenTenRequest {
  content: string; // 평문
}

export interface TenTenWithTopic extends TenTen {
  topicDate: string;
  topicDescription?: string;
}

export interface DecryptedTenTen extends Omit<TenTen, 'content'> {
  content: string; // 복호화된 평문
}

